Src/IronPython/Compiler/Ast/PythonAst.cs

```diff
@@ -815,14 +822,27 @@ internal class LookupVisitor : MSAst.ExpressionVisitor {
 
                 ScopeStatement prevScope = _curScope;
                 try {
-                    // rewrite the comprehension in a new scope
+                    // mark the first (outermost) "for" iterator for rewrite in the current scope
+                    _outerComprehensionScopes[((ComprehensionFor)comprehension.Iterators[0]).List] = _curScope;
```

Comment:
The first element of `Iterators` is always a `ComprehensionFor`? Probably need to revisit that Ast at some point...

Reply:
Yes, the first element of `Iterators` is always a `ComprehensionFor`. Furthermore, there is always at least one element in `Iterators`. The [parser already ensures it](https://github.com/IronLanguages/ironpython3/blob/8b853833aef23d327c64f4655d78152cfc05df24/Src/IronPython/Compiler/Parser.cs#L2684) (ignore the grammar comments above the methods, they do not exactly describe what the methods do). Also `ast` [organizes its "iterators"](https://github.com/IronLanguages/ironpython3/blob/8b853833aef23d327c64f4655d78152cfc05df24/Src/IronPython/Modules/_ast.cs#L580) as a sort of list of lists, each sublist headed by one `ComprehensionFor` and followed by zero or more `ComprehensionIf`.

For peace of mind, maybe a debug assert in the comprehension constructor?